server/pbench/bin/pbench-audit-archive

```diff
@@ -0,0 +1,200 @@
+    if [ -s ${unexpectedfiles} ]; then
+        printf "\t* Unexpected files in controller directory:\n"
+        printf "\t  ++++++++++\n"
+        cat ${unexpectedfiles}
+        printf "\t  ----------\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${unexpectedfiles}
+
+    find ${controller} -maxdepth 1 -type f \( -name '*.tar.xz.md5' -o -name '*.tar.xz' \) > ${tarballs} 2>&1
+    if [ ! -s ${tarballs} ]; then
+        printf "\t* No tar ball files found in this controller directory.\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${tarballs}
+
+    return $cnt
+}
+
+function verify_prefixes {
+    controller=${1}
+
+    if [ ! -e ${controller}/.prefix ]; then
+        return 0
+    fi
+    if [ ! -d ${controller}/.prefix ]; then
+        printf "\t* Prefix directory, .prefix, is not a directory\n"
+        return 1
+    fi
+
+    let cnt=0
+
+    find ${controller}/.prefix -maxdepth 1 ! -name 'prefix.*' ! -name '*.prefix' -printf "\t  %f\n" | sort > ${nonprefixes} 2>&1
+    if [ -s ${nonprefixes} ]; then
+        printf "\t* Unexpected file system objects in .prefix directory:\n"
+        printf "\t  ++++++++++\n"
+        cat ${nonprefixes}
+        printf "\t  ----------\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${nonprefixes}
+
+    find ${controller}/.prefix -maxdepth 1 -name 'prefix.*' -printf "\t  %f\n" | sort > ${wrongprefixes} 2>&1
+    if [ -s ${wrongprefixes} ]; then
+        printf "\t* Wrong prefix file names found in /.prefix directory:\n"
+        printf "\t  ++++++++++\n"
+        cat ${wrongprefixes}
+        printf "\t  ----------\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${wrongprefixes}
+
+    return $cnt
+}
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+# Initialize index mail content
+> $index_content
+
+# Construct the report file
+printf "\nstart-$(timestamp): archive hierarchy: $ARCHIVE\n" | tee > $report
+
+let ret=0
+
+# Find all the non-directory files at the same level of the controller
+# directories and report them, keeping them in sorted order by name.
+find $ARCHIVE -maxdepth 1 ! -type d  -printf "\t%M %s %t %f\n" | sort -k 8 > $badcontrollers
+if [ -s $badcontrollers ]; then
+    printf "\n\nBad Controllers:\n" >> $report
+    cat $badcontrollers >> $report
+    let ret=ret+1
+fi
+
+# Find all the normal controller directories, ignoring the "." (current)
+# directory (if the $ARCHIVE directory resolves to "."), and ignoring the
+# $ARCHIVE directory itself, while keeping them all in sorted order.
+find $ARCHIVE -maxdepth 1 -type d ! -name . ! -name $(basename $ARCHIVE) | sort > $controllers
+while read controller ;do
+    lclreport="${workdir}/$(basename ${controller})"
+    > $lclreport
+    verify_subdirs $controller >> $lclreport
+    verify_tarball_names $controller >> $lclreport
+    verify_prefixes $controller >> $lclreport
+    if [ -s $lclreport ]; then
+        printf "\n\nController: $(basename ${controller})\n" >> $report
```

Comment:
Maybe this is enough and you can disregard the previous comments.

Reply:
Was hoping to avoid duplicating information in the report.